src/utils/spo.ts

```diff
@@ -1453,5 +1456,213 @@ export const spo = {
     const webProperties: WebProperties = await request.get<WebProperties>(requestOptions);
 
     return webProperties;
+  },
+
+  /**
+  * Gets file properties by server relative file url
+  * Returns a ListItemAllFields object
+  * @param webUrl Web url
+  * @param url The url of the list
+  * @param logger The logger object
+  * @param verbose If in verbose mode
+  */
+  async getFileAsListItemByUrl(webUrl: string, url: string, logger?: Logger, verbose?: boolean): Promise<any> {
+    if (verbose && logger) {
+      logger.logToStderr(`Getting the file properties with url ${url}`);
+    }
+
+    let requestUrl = `${webUrl}/_api/web/GetFileByServerRelativePath(DecodedUrl=@f)?$expand=ListItemAllFields`;
+    const serverRelativePath = urlUtil.getServerRelativePath(webUrl, url);
+    requestUrl += `&@f='${formatting.encodeQueryParameter(serverRelativePath)}'`;
+
+    const requestOptions: CliRequestOptions = {
+      url: requestUrl,
+      headers: {
+        'accept': 'application/json;odata=nometadata'
+      },
+
+      responseType: 'json'
+    };
+
+    const file = await request.get<any>(requestOptions);
+
+    return file.ListItemAllFields;
+  },
+
+  /**
+  * Updates a list item
+  * Returns the updated listitem object
+  * @param webUrl Web url
+  * @param listUrl The url of the list
+  * @param id The id of the list item
+  * @param systemUpdate If the item should be updated with system update
+  * @param options an object of the properties that should be updated
+  * @param logger The logger object
+  * @param verbose If in verbose mode
+  */
+  async setListItem(webUrl: string, listUrl: string, id: string, systemUpdate?: boolean, options?: object, logger?: Logger, verbose?: boolean): Promise<any> {
```

Comment:
in principle, this method seems too complicated for an util. In general, the util should be small and reusable. This method seems quite large and specialized. 
Also we should research if now we may reuse this in `spo listitem set` command in order to avoid duplicates. I guess it won't be that easy but maybe splitting this to smaller peaces will make it possible